src/systask_with_env.c

```diff
@@ -0,0 +1,28 @@
+#include "global.h"
```

Comment:
systask_with_env sounds too wordy but I cant think of a better one so as long as @red031000 doesn't have an issue with it this file looks good to me

Reply:
systask_environment.c imo
and change Env to Environment generally, would keep it fairly consistent with others